partner_contact_age_range/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    age_range_id = fields.Many2one(
+        "res.partner.age.range", "Age Range", compute="_compute_age_range", store=True
+    )
+
+    @api.depends("age")
+    def _compute_age_range(self):
+        for record in self:
+            age_range = self.env["res.partner.age.range"].search(
+                [("age_from", "<=", record.age), ("age_to", ">=", record.age)]
+            )
+            if age_range:
+                record.age_range_id = age_range.id
+
+    def _update_age_range(self):
+        """
+        This method is called from a cron job.
+        It is used to update age range on contact
+        """
+        records = self.search([("birthdate_date", "!=", False)])
+        for partner in records:
+            partner._compute_age_range()
```

Comment:
```suggestion
        partners = self.search([("birthdate_date", "!=", False)])
        partners._compute_age_range()
```
This way you benefict from odoo caching and it's also cleaner